### PR TITLE
fs: try to fix a deadlock in the new code

### PR DESCRIPTION
**Pull request by maruel**
Remove the lock from wakeUpLoop(). I suspect it is what was deadlocking
on travis.

Remove the tear down code. It was hard to make tear down never deadlock in unit
tests, so leak the goroutine instead.

Properly release the lock on initialization failure.

Make follow up init() call to be lockless.

### CODE REVIEW

File: host/fs/fs_linux.go [hunk 1]
```diff
@@ -164,64 +164,34 @@ func (e *eventsListener) init() error {
...
 	e.wakeUp = wakeUp
-	// The mutex is still held after this function exits, it's loop() that will
-	// release the mutex.
+	e.fds = map[int32]chan<- time.Time{}
+	e.fds[int32(e.r.Fd())] = wakeUp
+	// The mutexes are still held after this function exits, it's loop() that will
+	// release the mutexes.
 	//
 	// This forces loop() to be started before addFd() can be called by users.
 	e.wg.Add(1)
```
Comment (NeuralSpaz) [inline host/fs/fs_linux.go]: nothing waits for this wait group.

File: host/fs/fs_linux.go [hunk 2]
```diff
@@ -164,64 +164,34 @@ func (e *eventsListener) init() error {
...
-	e.r = nil
-	err := syscall.Close(e.epollFd)
-	e.epollFd = 0
-	e.mu.Unlock()
-	return err
-}
-
 // loop is the main event loop.
 func (e *eventsListener) loop() {
 	defer e.wg.Done()
```
Comment (NeuralSpaz) [inline host/fs/fs_linux.go]: same here, nothing is waiting.

File: host/fs/fs_linux.go [hunk 3]
```diff
@@ -136,26 +136,26 @@ func (e *event) wait(timeoutms int) (int, error) {
 type eventsListener struct {
 	wg sync.WaitGroup
```
Comment (NeuralSpaz) [inline host/fs/fs_linux.go]: no longer used.

File: host/fs/fs_linux.go (hunk 1 above)
Comment (maruel) [reply host/fs/fs_linux.go]: Thanks for catching this, removed the WaitGroup.

File: host/fs/fs_linux.go (hunk 3 above)
Comment (maruel) [reply host/fs/fs_linux.go]: Removed.

File: host/fs/fs_linux.go (hunk 2 above)
Comment (maruel) [reply host/fs/fs_linux.go]: Removed.

**Pull request merged**